modules/oe_content_project/src/Commands/CopyBudgetValues.php

```diff
@@ -0,0 +1,172 @@
+   */
+  protected $messenger;
+
+  /**
+   * The node storage.
+   *
+   * @var \Drupal\node\NodeStorageInterface
+   */
+  protected $nodestorage;
+
+  /**
+   * UpdatePersonData class constructor.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, MessengerInterface $messenger) {
+    parent::__construct();
+
+    $this->entityTypeManager = $entityTypeManager;
+    $this->messenger = $messenger;
+    $this->nodeStorage = $this->entityTypeManager->getStorage('node');
+  }
+
+  /**
+   * Triggers the field value copy.
+   *
+   * @command project-budget-copy-values:run
+   */
+  public function copyFieldValues(): void {
+    $project_ids = $this->nodeStorage->getQuery()
+      ->condition('type', 'oe_project')
+      ->latestRevision()
+      ->sort('nid')
+      ->accessCheck(FALSE)
+      ->execute();
+
+    if (!$project_ids) {
+      return;
+    }
+
+    $batch_builder = (new BatchBuilder())
+      ->setTitle($this->t('Copy project budget field values'))
+      ->setFinishCallback([$this, 'processFinished']);
+    $batch_builder->addOperation([$this, 'processUpdate'], [
+      $project_ids,
+    ]);
+    batch_set($batch_builder->toArray());
+    drush_backend_batch_process();
+  }
+
+  /**
+   * Batch operation to process the nodes.
+   *
+   * @param array $project_ids
+   *   The project node IDs.
+   * @param array|DrushBatchContext $context
+   *   The batch context.
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   * @SuppressWarnings(PHPMD.NPathComplexity)
+   */
+  public function processUpdate(array $project_ids, &$context): void {
+    if (!isset($context['results']['processed'])) {
+      $context['results']['processed'] = [];
+      $context['results']['failed'] = [];
+    }
+
+    $sandbox = &$context['sandbox'];
+    if (!$sandbox) {
+      $sandbox['progress'] = 0;
+      $sandbox['project_ids'] = array_unique($project_ids);
+      $sandbox['max'] = count($sandbox['project_ids']);
+    }
+
+    // Process the project nodes.
+    if ($sandbox['project_ids']) {
+      $id = array_pop($sandbox['project_ids']);
+      $this->updateNode($id);
+      $context['results']['processed'][] = $id;
+    }
+
+    $sandbox['progress']++;
+
+    $context['finished'] = $sandbox['progress'] / $sandbox['max'];
+  }
+
+  /**
+   * Callback for when the batch processing completes.
+   *
+   * @param bool $success
+   *   Whether the batch was successful.
+   * @param array $results
+   *   The batch results.
+   */
+  public function processFinished($success, array $results): void {
+    if (!$success) {
+      $this->messenger->addError($this->t('There was a problem with the batch.'));
+      return;
+    }
+
+    $processed = count($results['processed']);
+    if ($processed === 0) {
+      $this->messenger->addStatus($this->t('No nodes have been processed.'));
+    }
+    else {
+      $this->messenger->addStatus($this->t('@items nodes with values have been processed: @ids.', [
```

Comment:
Are we sure we want to list all the node IDs that we updated? What if we have one thousand nodes?
Anyway fine by me.